mesmer/calibrate_mesmer/__init__.py

```diff
@@ -1,11 +1,349 @@
+    def __init__(
+        self,
+        esms,
+        scenarios,
+        cmip_generation,
+        cmip_data_root_dir,
+        observations_root_dir,
+        auxiliary_data_dir,
+        reference_period_type,
+        reference_period_start_year,
+        reference_period_end_year,
+        ens_type_tr,
+        tas_global_trend_method,
+        hfds_global_trend_method,
+        tas_global_variability_method,
+        tas_local_trend_method,
+        tas_local_variability_method,
+        method_lt_each_gp_sep,
+        nr_emus_v,
+        seeds,
+        weight_scenarios_equally,
+        threshold_land,
+        cross_validation_max_iterations,
+    ):
+        self.esms = esms
+        self.scenarios = scenarios
+        self.gen = cmip_generation
+        # TODO: remove need for trailing seperator eventually
+        self.dir_cmipng = f"{cmip_data_root_dir}{os.sep}"
+        self.dir_obs = f"{observations_root_dir}{os.sep}"
+        self.dir_aux = f"{auxiliary_data_dir}{os.sep}"
+        self.ref = {
+            "type": reference_period_type,
+            "start": reference_period_start_year,
+            "end": reference_period_end_year,
+        }
+        self.ens_type_tr = ens_type_tr
+        self.methods = {
+            "tas": {
+                "gt": tas_global_trend_method,
+                "gv": tas_global_variability_method,
+                "lt": tas_local_trend_method,
+                "lv": tas_local_variability_method,
+            },
+            "hfds": {"gt": hfds_global_trend_method},
+        }
+        self.method_lt_each_gp_sep = method_lt_each_gp_sep
+
+        # TODO: should this be hard-coded like this?
+        self.preds = {
+            "tas": {
+                "gt": ["saod"],
+                "gv": [],
+            },
+            "hfds": {"gt": []},
+        }
+        self.preds["tas"]["g_all"] = self.preds["tas"]["gt"] + self.preds["tas"]["gv"]
+
+        self.nr_emus_v = nr_emus_v
+        self.seed = seeds
+        self.wgt_scen_tr_eq = weight_scenarios_equally
+        self.threshold_land = threshold_land
+        self.max_iter_cv = cross_validation_max_iterations
+
+
+def _calibrate_mesmer(
+    esms,
+    scenarios_to_train,
+    target_variable,
+    reg_type,
+    threshold_land,
+    output_file,
+    scen_seed_offset_v,
+    cmip_data_root_dir,
+    observations_root_dir,
+    auxiliary_data_dir,
+    cmip_generation=6,
+    reference_period_type="individ",
+    reference_period_start_year="1850",
+    reference_period_end_year="1900",
+    ens_type_tr="msic",  # TODO: work out what this is
+    tas_global_trend_method="LOWESS_OLSVOLC",
+    hfds_global_trend_method="LOWESS",
+    tas_global_variability_method="AR",
+    tas_local_trend_method="OLS",
+    tas_local_variability_method="OLS_AR1_sci",
+    method_lt_each_gp_sep=True,  # TODO: work out what this is
+    nr_emus_v=100,  # Why is this required now?
+    weight_scenarios_equally=True,
+    cross_validation_max_iterations=30,
+):
+
```

Comment:
```suggestion
    """calibrate mesmer - additional predictors configuration. used for end-to-end test"""
```
or something like this

Reply:
we would ideally write this so it's easier for the user to choose what to use as a predictor...

Reply:
Update as part of https://github.com/MESMER-group/mesmer/issues/19